### ISSUE TITLE
Predicting posterior probabilities of new data with gridsearch

### ISSUE THREAD

**Issue by olsen378** [CecileProust-Lima/lcmm#75]
Hi Cécile,

I'm using your solution to issue #59 to extract the posterior probabilities for a held out test data set for a latent class model (thanks for that!). My code for the 1 and 4 class training and testing models is below. My accuracy for the testing set is extremely high. Further, for the one class model, the model$best for the training and testing sets match, and the marginal predictions match those when using the predicty function. However, for the 4 class model the model$best for training and testing are different, and the marginal predictions don't match those when using the predicty function (indicating the the model is actually being re-fit for the testing data?). Any idea why that might be the case? Could it be because I am using the gridsearch function for the training models? If so, I have tried using gridsearch in the testing model, and haven't been able to come up with a solution that will make the model$bests the same for the training and testing models, or give me the same marginal predictions as predicty - Do you have a solution for this? Additionally, is there a way to tell whether or not the 4 class model is actually being re-fit to the testing data, other than the ways I mentioned above?

Training models: 
 train_week_gmm3_1<-hlme(PHQ9_SCORE~ Week_Num + I(Week_Num^2) + I(Week_Num^3), subject = "PID",random=~Week_Num+ I(Week_Num^2) + I(Week_Num^3), ng = 1, data = train)

train_week_gmm3_4<-gridsearch(rep = 100, maxiter = 30, minit = train_week_gmm3_1, hlme(PHQ9_SCORE~ Week_Num + I(Week_Num^2) + I(Week_Num^3), subject = "PID",random=~Week_Num + I(Week_Num^2) + I(Week_Num^3),ng = 4, data = train, mixture = ~ Week_Num + I(Week_Num^2) + I(Week_Num^3), nwg=TRUE))

Testing models:
test_week_gmm3_1 <- hlme(PHQ9_SCORE~ Week_Num + I(Week_Num^2)+ I(Week_Num^3), subject = "PID",random=~Week_Num + I(Week_Num^2)+ I(Week_Num^3),ng = 1, data = test, nwg=FALSE, B=train_week_gmm3_1$best, maxiter = 0)

test_week_gmm3_4 <- hlme(PHQ9_SCORE~ Week_Num + I(Week_Num^2)+ I(Week_Num^3), subject = "PID",random=~Week_Num + I(Week_Num^2)+ I(Week_Num^3),ng = 4, data = test, mixture = ~ Week_Num + I(Week_Num^2)+ I(Week_Num^3), nwg=TRUE, B=train_week_gmm3_4$best, maxiter = 0)

I look forward to hearing from you, and greatly appreciate any insight you have.

-Sarah

**Comment by CecileProust-Lima** [comment]
Hi Sarah, 
if you use maxiter=0 in the test set and B=your estimates from the training set, then the program does not update the parameter estimates, it only computes the outputs at the B parameter values, that is the fitted values, the posterior probabilities, the random-effects predictions, etc for the test set. 
This is weird that for the 4 class model the model$best for training and testing are different IF you use maxiter=0 for the testing. If you could replicate your problem with publicly available data (let's say the paquid sample in lcmm), that would help. We will also have a look. 
Best
Cécile

**Comment by olsen378** [comment]
Thanks so much for your response! 

The model$best for the 4 class model was only different when I was using gridsearch. Without gridsearch for the multi-class model, model$best is the same.

**Comment by CecileProust-Lima** [comment]
We will look into this. 
Thanks for the feedback. 
Cécile
